src/mixins/I18n.ts

```diff
@@ -72,21 +78,33 @@ export interface I18nMixin {
 	properties: I18nProperties;
 }
 
+export function registerI18nInjector(localeData: LocaleData, registry: Registry): Injector {
+	const injector = new Injector(localeData);
+	registry.defineInjector(INJECTOR_KEY, injector);
+	return injector;
+}
+
 export function I18nMixin<T extends Constructor<WidgetBase<any>>>(Base: T): T & Constructor<I18nMixin> {
-	class I18n extends Base {
+	@inject({
+		name: INJECTOR_KEY,
+		getProperties: (localeData: LocaleData, properties: I18nProperties) => {
+			const { locale, rtl } = localeData;
+			const injected: Partial<I18nProperties> = {};
```

Comment:
Does this need to be a Partial?

Reply:
would something like this work?
```ts
getProperties: (localeData: LocaleData, properties: I18nProperties) => {
    const { locale = localData.locale, rtl = localData.rtl } = properties;

    return {
        locale,
        rtl
    };
}
```

Reply:
Yes, that would work. I've noticed elsewhere that an empty object is returned if nothing needs to be changed, so I kept that here.